tree/20_cfe_basics/dry_run_lib.cf

```diff
@@ -34,6 +54,11 @@
 # WARNING: * promise type specific dry-run is NOT considered by action bodies, and will be overriden
 # WARNING:   when using any action body.
 # WARNING:
+#
+# @parameter mode   The dry-run mode, "true" to enable dry-run-mode and "false" to disable it
+# @agent_requirements "capabilities": [ "defaultbody" ] || "agent_version"" >= 3.9
```

Comment:
It was not implemented because it is very long and complex to implement in standard shell, I hope we can find time to implement it before the end of times